app/scripts/lib/rpc-method-middleware/handlers/wallet-requestPermissions.ts

```diff
@@ -105,65 +98,54 @@ async function requestPermissionsImplementation(
 
   const hasCaip25EquivalentPermissions =
     Object.keys(caip25EquivalentPermissions).length > 0;
-  const hasOtherRequestedPermissions =
-    Object.keys(requestedPermissions).length > 0;
-
-  let grantedPermissions: GrantedPermissions = {};
 
-  let caip25Approval;
   if (hasCaip25EquivalentPermissions) {
-    try {
-      caip25Approval = await requestCaip25ApprovalForOrigin(
-        caip25EquivalentPermissions,
-      );
-    } catch (error) {
-      return end(error as unknown as Error);
-    }
+    const caip25Permission = getCaip25PermissionFromLegacyPermissionsForOrigin(
+      caip25EquivalentPermissions,
+    );
+    requestedPermissions = { ...requestedPermissions, ...caip25Permission };
   }
 
-  if (hasOtherRequestedPermissions || !hasCaip25EquivalentPermissions) {
-    try {
-      const [frozenGrantedPermissions] = await requestPermissionsForOrigin(
-        requestedPermissions,
-      );
-      grantedPermissions = { ...frozenGrantedPermissions };
-    } catch (error) {
-      return end(error as unknown as Error);
-    }
-  }
+  let grantedPermissions: GrantedPermissions = {};
+
+  const [frozenGrantedPermissions] = await requestPermissionsForOrigin(
```

Comment:
Why can we now always call `requestPermissionsForOrigin` regardless of what `requestedPermissions` is?

Reply:
Because the caip25 permission no longer needs to be handled specially since we rely on the factory method directly now. we translate any requested eth_accounts and permitted chains permission into a caip25 permission and then directly request permission for it

Reply:
Gotcha!